src/email/email.js

```diff
@@ -0,0 +1,120 @@
+const { SMTPServer } = require("smtp-server");
+const { MailParser } = require('mailparser');
+const ParseEmailAddress = require("email-addresses");
+const { Logging } = require('../log');
+
+const log = Logging.get("email");
+
+const roomAlias = function(rcptTo, mxDomain) {
+    let localpartRcptTo;
+    for (let i = 0; i < rcptTo.length; i++) {
+        localpartRcptTo = ParseEmailAddress.parseOneAddress(rcptTo[i].address).local;
+        if ( localpartRcptTo.endsWith(mxDomain)) {
+            log.info("Message destination address:", localpartRcptTo);
+            break;
+        }
+    }
+    return getUserIdorAlias(localpartRcptTo);
```

Comment:
It might be easier to frame your logic like this?

```suggestion
            log.info("Message destination address:", localpartRcptTo);
            return getUserIdorAlias(localpartRcptTo);
        }
    }
    // Throw if not found
    throw Error("Could not determine alias from address");
```